### PR TITLE
Provide json file dependency for user feature

### LINKED ISSUE #667: user feature install fails with Gradle 7
Running `gradle installFeature` to install user feature fails on Gradle 7.

Related Gradle issue:
https://github.com/gradle/gradle/issues/13122

Problem happens when we recompile Gradle using the ProjectBuilder API to fetch new features.json artifact (generated by prepareFeature) from the Mavenlocal repo. 

==Error log & stack trace==
* What went wrong:
Execution failed for task ':installFeature'.
> The settings are not yet available for build 'regFeature'.
...
Caused by: java.lang.IllegalStateException: The settings are not yet available for build 'regFeature'.
        at org.gradle.invocation.DefaultGradle.getSettings(DefaultGradle.java:207)
        at org.gradle.invocation.DefaultGradle_Decorated.getSettings(Unknown Source)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.ignoresSettingsRepositories(DefaultConfiguration.java:756)

### PR DESCRIPTION
**Pull request by jjiwooLim**
fixes #667
Instead of creating a new instance of gradle, use the file dependency configuration from prepareFeature task

### CODE REVIEW

File: src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractFeatureTask.groovy [hunk 1]
```diff
@@ -38,10 +39,10 @@ public class AbstractFeatureTask extends AbstractServerTask {
 
     private InstallFeatureUtil util;
 
-    @Internal
-    Project newProject = project;
-
     private ServerFeatureUtil servUtil;
+	
+	@Internal
+ 	def jsonCoordinate;
```
Comment (mattbsox) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractFeatureTask.groovy:45]: Could change `def` to a String and update the first part of the check on 134 to null and empty checks.

**Comment by jjiwooLim** [comment]
Restored gradle.yml to use ci.common main branch

File: src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractFeatureTask.groovy (hunk 1 above)
Comment (cherylking) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractFeatureTask.groovy:45]: I think you missed Matt's point. You would still check if it is equal, but first check that `jsonCoordinate != null and !jsonCoordinate.empty() && coordinates.equals(jsonCoordinate)`

**Review by mattbsox — APPROVED** [review]

**Comment by cherylking** [comment]
@jjiwooLim Please do not merge this until you resolve the compilation error:
```
/home/runner/work/ci.gradle/ci.gradle/src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractPrepareTask.groovy: 89: Method 'provideJsonFileDependency' from class 'io.openliberty.tools.gradle.tasks.AbstractPrepareTask$PrepareFeatureTaskUtil' does not override method from its superclass or interfaces but is annotated with @Override.
 @ line 89, column 3.
   		@Override
```

**Comment by jjiwooLim** [comment]
> @jjiwooLim Please do not merge this until you resolve the compilation error:
> 
> ```
> /home/runner/work/ci.gradle/ci.gradle/src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractPrepareTask.groovy: 89: Method 'provideJsonFileDependency' from class 'io.openliberty.tools.gradle.tasks.AbstractPrepareTask$PrepareFeatureTaskUtil' does not override method from its superclass or interfaces but is annotated with @Override.
>  @ line 89, column 3.
>    		@Override
> ```

After merging ci.common PR, the compilation errors were resolved.

**Pull request merged**